### PR TITLE
Fix/reduce rerenders due to flow data api

### PR DESCRIPTION
**Pull request by abhijitb**
Flushing the queue on all steps to make sure that the store is in sync with the database.

### CODE REVIEW

**Review by officiallygod — COMMENTED** [review]
Sometimes I was getting the wrong headers.

https://github.com/newfold-labs/wp-module-onboarding/assets/48691514/da8cf42f-0e98-47b8-a826-049e8b205a86

File: src/OnboardingSPA/pages/Steps/DesignColors/index.js [hunk 1]
```diff
@@ -50,7 +54,10 @@ const StepDesignColors = () => {
 	};
 
 	useEffect( () => {
-		if ( THEME_STATUS_ACTIVE === themeStatus ) getStylesAndPatterns();
+		if ( THEME_STATUS_ACTIVE === themeStatus ) {
+			flushQueue();
```
Comment (officiallygod) [inline src/OnboardingSPA/pages/Steps/DesignColors/index.js:58]: If we could somehow await this and check the queue if it is empty before we make a request won't that work?

**Review by officiallygod — COMMENTED** [review]
As I am still running into the same issue of getting an older header after selection.


https://github.com/newfold-labs/wp-module-onboarding/assets/48691514/5a4e295e-e5b9-4471-9243-d45a74671b6a

**Review by officiallygod — COMMENTED** [review]
Now I am getting a GET as well as POST request on every click. Am I doing anything wrong?

https://github.com/newfold-labs/wp-module-onboarding/assets/48691514/b92c7921-8863-4cdb-894e-ecb4cdff6a1b

**Comment by abhijitb** [comment]
After making several attempts to modify and optimize the code it looks like the current implementation is sufficient hence abandoning this PR.

**Pull request closed without merging**